### PR TITLE
Fixing pulling of security plugin to run integ tests in secure opensearch cluster

### PR DESCRIPTION
**Pull request by vibrantvarun**
### Description
The PR fixes the security plugin issue in the neural search build.

Premise:
While building neural search we are pulling security plugin via zip archive.
```
    secureIntegTestPluginArchive group: 'org.opensearch.plugin', name:'opensearch-security', version: "${opensearch_build}"
```

But in neural search we do not use security plugin as a compile time dependency.

How zipArchive works?
Zip Archive works by resolving all the dependencies provided in the dependencies section. opensearch-knn we use at compile time dependency and therefore we extract it as a jar in task extractKNN Jar. opensearch-ml-plugin zip we use it in bwc tests.

```
// In order to add the jar to the classpath, we need to unzip the
// k-NN zip and then copy it into a directory that we specify as a dependency.
task extractKnnJar(type: Copy) {
    mustRunAfter()
    from(zipTree(configurations.zipArchive.find { it.name.startsWith("opensearch-knn")}))
    into knnJarDirectory
}
```

Now this line(from(zipTree(configurations.zipArchive.find { it.name.startsWith("opensearch-knn")})) ) of code tries to find opensearch-security as well however we don't use it.

The fix:

Segregate the pulling of security plugin in secureIntegTestPluginArchive configuration. 

Also, 
In this PR, Java 21 has been restricted to 21.0.1 due to ongoing issue in 21.0.2 version.
https://github.com/opensearch-project/OpenSearch/pull/11968#pullrequestreview-1837589496.
Once the fix to the issue will be merged, will revert back to pulling of Java 21 to latest version.

Moreover, The bwc tests are disabled in the PR due to there is an ongoing failure of security tests due to zip archive issue. Therefore the latest build has not been published from last few days and bwc tests are failing.

### Check List
- [x] New functionality includes testing.
    - [x] All tests pass
- [x] New functionality has been documented.
    - [x] New functionality has javadoc added
- [x] Commits are signed as per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/k-NN/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by vibrantvarun** [comment]
BWC tests failure is expected as we do not have latest neural-search 2.12 artifact.

File: build.gradle [hunk 1]
```diff
@@ -160,6 +160,7 @@ allprojects {
 
 configurations {
     zipArchive
+    secureIntegTestPluginArchive
```
Comment (jmazanec15) [inline build.gradle:163]: In the future, I think we could make the configurations follow an inheritance hierarchy:
```
zipArchive
compileTimeZipArchive.extendsFrom zipArchive // for k-NN
```

but I think for now this is fine.

**Review by ryanbogan — APPROVED** [review]

**Review by martin-gaievski — APPROVED** [review]

File: .github/workflows/backwards_compatibility_tests_workflow.yml [hunk 2]
```diff
@@ -35,7 +36,7 @@ jobs:
       - name: Run NeuralSearch Restart-Upgrade BWC Tests from BWCVersion-${{ matrix.bwc_version }} to OpenSearch Version-${{ matrix.opensearch_version }} on ${{matrix.os}}
         run: |
           echo "Running restart-upgrade backwards compatibility tests ..."
-          ./gradlew :qa:restart-upgrade:testAgainstNewCluster -D'tests.bwc.version=${{ matrix.bwc_version }}'
+#          ./gradlew :qa:restart-upgrade:testAgainstNewCluster -D'tests.bwc.version=${{ matrix.bwc_version }}'
```
Comment (heemin32) [inline .github/workflows/backwards_compatibility_tests_workflow.yml:39]: Could you leave a comment why we are commenting this line out?

Comment (vibrantvarun) [reply .github/workflows/backwards_compatibility_tests_workflow.yml:39]: sure

**Review by heemin32 — APPROVED** [review]

**Pull request merged**